lib/codegen/src/predicates.rs

```diff
@@ -11,6 +11,20 @@
 
 use ir;
 
+/// Check that the `x` is zero, regardless of sign bit.
+#[allow(dead_code)]
+pub fn is_zero_64_bit_float<T: Into<ir::immediates::Ieee64>>(x: T) -> bool {
+    let x64 = x.into();
+    (x64.bits() << 1) == 0
```

Comment:
I apologize for being unclear; the predicate we need here is `x64.bits() == 0`, because even though -0 is equal to 0, it's a different bit pattern, and it matters in some cases, so we can only use the xorps/xorpd optimization for plain 0, and not -0. And same for the f32 case below.